nano/secure/store_partial.hpp

```diff
@@ -0,0 +1,217 @@
+		nano::pruned_store_partial<Val, Derived_Store> & pruned_store_partial_a,
+		nano::peer_store_partial<Val, Derived_Store> & peer_store_partial_a,
+		nano::confirmation_height_store_partial<Val, Derived_Store> & confirmation_height_store_partial_a,
+		nano::final_vote_store_partial<Val, Derived_Store> & final_vote_store_partial_a,
+		nano::version_store_partial<Val, Derived_Store> & version_store_partial_a) :
+		store{
+			block_store_partial_a,
+			frontier_store_partial_a,
+			account_store_partial_a,
+			pending_store_partial_a,
+			unchecked_store_partial_a,
+			online_weight_store_partial_a,
+			pruned_store_partial_a,
+			peer_store_partial_a,
+			confirmation_height_store_partial_a,
+			final_vote_store_partial_a,
+			version_store_partial_a
+		}
+	{}
+	// clang-format on
+
+	/**
+	 * If using a different store version than the latest then you may need
+	 * to modify some of the objects in the store to be appropriate for the version before an upgrade.
+	 */
+	void initialize (nano::write_transaction const & transaction_a, nano::genesis const & genesis_a, nano::ledger_cache & ledger_cache_a) override
+	{
+		auto hash_l (genesis_a.hash ());
+		debug_assert (account.begin (transaction_a) == account.end ());
+		genesis_a.open->sideband_set (nano::block_sideband (network_params.ledger.genesis_account, 0, network_params.ledger.genesis_amount, 1, nano::seconds_since_epoch (), nano::epoch::epoch_0, false, false, false, nano::epoch::epoch_0));
+		block.put (transaction_a, hash_l, *genesis_a.open);
+		++ledger_cache_a.block_count;
+		confirmation_height.put (transaction_a, network_params.ledger.genesis_account, nano::confirmation_height_info{ 1, genesis_a.hash () });
+		++ledger_cache_a.cemented_count;
+		ledger_cache_a.final_votes_confirmation_canary = (network_params.ledger.final_votes_canary_account == network_params.ledger.genesis_account && 1 >= network_params.ledger.final_votes_canary_height);
+		account.put (transaction_a, network_params.ledger.genesis_account, { hash_l, network_params.ledger.genesis_account, genesis_a.open->hash (), std::numeric_limits<nano::uint128_t>::max (), nano::seconds_since_epoch (), 1, nano::epoch::epoch_0 });
+		++ledger_cache_a.account_count;
+		ledger_cache_a.rep_weights.representation_put (network_params.ledger.genesis_account, std::numeric_limits<nano::uint128_t>::max ());
+		frontier.put (transaction_a, hash_l, network_params.ledger.genesis_account);
+	}
+
+	bool root_exists (nano::transaction const & transaction_a, nano::root const & root_a) override
+	{
+		return block.exists (transaction_a, root_a.as_block_hash ()) || account.exists (transaction_a, root_a.as_account ());
+	}
+
+	bool exists (nano::transaction const & transaction_a, tables table_a, nano::db_val<Val> const & key_a) const
+	{
+		return static_cast<const Derived_Store &> (*this).exists (transaction_a, table_a, key_a);
+	}
+
+	int const minimum_version{ 14 };
+
+protected:
+	nano::network_params network_params;
+	int const version_number{ 21 };
```

Comment:
Because it was conflicting with version store, changed the variable name to version_number.